Rakefile

```diff
@@ -8,7 +8,7 @@ require "rake/testtask"
 Rake::TestTask.new(:spec) do |t|
   t.libs << "lib"
   t.ruby_opts << "-r./spec/spec_helper"
-  t.warning = false # TODO: true
+  t.warning = !!ENV['RUBY_WARNINGS_ENABLED']
```

Comment:
I think it would be preferable for the warnings to be enabled if _either_ the environment variable is set to true or the environment variable is unset. I think we ideally want these warnings to display and to only hide them if the user has explicitly said otherwise.

Reply:
Another thing to consider would be to tie into Ruby's own warning settings.

```
RUBYOPT=-w rake
```

Then

```
t.warning = !!$VERBOSE
```

Granted, this doesn't give you warnings by default (and enables them for more than just the test suite).

If you did want to enable it by default, then I would go with something like `SUPPRESS_SPEC_WARNINGS` to disable it (I don't really think the RUBY qualifier is needed in this context).

Reply:
I see.. IMO, we can enable that once the warnings have been reduced to a minimal amount. Right now, the noise can muffle errors that should be handled.

Reply:
In that case, you might consider starting with my first suggestion.

Reply:
@mojavelinux This branch has maintainer access. Will you be able to push the necessary change?

Reply:
@ashmaroli The current maintainers are Jeanine, Goro, Drew and me. Dan doesn't have access to make the change.

Reply:
Ah! I didn't know that..